File: 2552-maximum-sum-of-distinct-subarrays-with-length-k/2552-maximum-sum-of-distinct-subarrays-with-length-k.py
class Solution:
    def maximumSubarraySum(self, nums: List[int], k: int) -> int:
        sets = set()
        sums = 0
        maxs = 0
        left = 0
        for right in range(0, len(nums)):
            while nums[right] in sets:
                sums = sums - nums[left]
                sets.remove(nums[left])
                left+=1
            sets.add(nums[right])
            sums = sums + nums[right]

            while right- left + 1> k:
                sums = sums - nums[left]
                sets.remove(nums[left])
                left+=1
            if right - left +1 == k:
                maxs = max(maxs,sums)
        return maxs

File: 2552-maximum-sum-of-distinct-subarrays-with-length-k/2552-maximum-sum-of-distinct-subarrays-with-length-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maximumSubarraySum = function (nums, k) {
    if (k == 1 && nums.length > 0) return Math.max(...nums)
    let set = new Set()
    let max = 0
    let sum = 0
    let left = 0
    for (let right = 0; right < nums.length; right++) {
        while (set.has(nums[right])) {
            set.delete(nums[left]);
            sum -= nums[left];
            left++;
        }
        set.add(nums[right])
        sum = sum + nums[right]

        while (right - left + 1 > k) {
            sum = sum - nums[left]
            set.delete(nums[left])
            left++
        }
        if (right - left + 1 == k) {
            max = Math.max(max, sum)
        }

    }
    return max
};